PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningHallWidgetAdapter.kt

```diff
@@ -0,0 +1,156 @@
+
+class DiningHallWidgetAdapter : RemoteViewsService(){
+
+    override fun onGetViewFactory(p0: Intent): RemoteViewsFactory {
+        return diningWidgetFactory(applicationContext, p0)
+    }
+
+    class diningWidgetFactory(private val context: Context, intent: Intent) : RemoteViewsFactory {
+        private var mDiningRequest: DiningRequest? = null
+        private var appWidgetId: Int = intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, AppWidgetManager.INVALID_APPWIDGET_ID)
+        private var dataSet: List<DiningHall> = emptyList()
+
+        // Connection to data source
+        override fun onCreate() {
+            //connect to data source
+            mDiningRequest = DiningHallWidget.diningRequestInstance
+            getWidgetDiningHalls()
+        }
+
+        // The place where we fetch data from the source set new data and update collection widget accordingly
+        override fun onDataSetChanged() {
+            //refresh data -> Update data every 30 minutes
+        }
+
+        override fun onDestroy() {
+            //close connection to data source
+        }
+
+        override fun getCount(): Int {
+            Log.d("msg", "${dataSet.size}")
+            return dataSet.size
+        }
+
+        override fun getViewAt(position: Int): RemoteViews {
+            val views = RemoteViews(context.packageName, R.layout.dining_hall_widget_item)
+            Log.d("msg", "position: ${position}, name: ${dataSet[position].name}")
+            views.setTextViewText(R.id.appwidget_text, dataSet[position].name)
+            views.setImageViewResource(R.id.diningBackground, dataSet[position].image)
+            if (dataSet[position].isOpen) {
+                Log.d("msg", "Open right now")
+                views.setImageViewResource(R.id.imageView, R.drawable.baseline_check_circle_24)
+                views.setInt(R.id.textView3, "setBackgroundColor", Color.parseColor("#6DB786"))
+                if (dataSet[position].openMeal() != "all" && dataSet[position].openMeal() != null) {
+                    val resources = context.resources
+                    val open_label : String = resources.getString(getOpenStatusLabel(dataSet[position].openMeal() ?: ""))
+                    views.setTextViewText(R.id.textView3, open_label)
+                }
+            }
+            else {
+                views.setImageViewResource(R.id.imageView, R.drawable.baseline_cancel_24)
+                views.setInt(R.id.textView3, "setBackgroundColor", Color.parseColor("#990000"))
+                if (dataSet[position].openTimes().isEmpty()) {
+                    views.setTextViewText(R.id.textView3, "Closed Today")
+                }
+                else {
+                    views.setTextViewText(R.id.textView3, "Closed")
+                }
+            }
+            val extras = Bundle()
+            extras.putInt("key_data", position)
+            val fillInIntent = Intent()
+            fillInIntent.putExtras(extras)
```

Comment:
Also do you actually use this data? I see in MainActivity.kt, you just switch to the dining tab, but you don't actually go to the selected dining hall. If you do not plan on using this, please remove it.